### ISSUE TITLE
Spawning 100+ processes causes segfault in libev

### ISSUE THREAD

**Issue by chriso** [joyent/libuv#287]
Dup of [node #2481](https://github.com/joyent/node/issues/2481) but posted here as it appears to be a libev issue.

**test.js**

``` javascript
var child = require('child_process');
for (var i = 0; i < 100; i++) {
    child.spawn('echo');
}
```

Running this on OS X Lion:

``` bash
~ $ node test.js
Segmentation fault: 11
```

Here's the GDB output:

```
~ $ gdb /tmp/node-v0.6.7/out/Release/node
GNU gdb 6.3.50-20050815 (Apple version gdb-1705) (Fri Jul  1 10:50:06 UTC 2011)
Copyright 2004 Free Software Foundation, Inc.
GDB is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.
Type "show copying" to see the conditions.
There is absolutely no warranty for GDB.  Type "show warranty" for details.
This GDB was configured as "x86_64-apple-darwin"...Reading symbols for shared libraries ........... done

(gdb) run test.js
Starting program: /private/tmp/node-v0.6.7/out/Release/node test.js
Reading symbols for shared libraries ++++++++++.......................................................................................................................... done

Program received signal EXC_BAD_ACCESS, Could not access memory.
Reason: KERN_PROTECTION_FAILURE at address: 0x0000000103dbd4b8
0x000000010026a8c9 in clear_pending (loop=0x1004ca3c0, w=0x100720b58) at ev.c:2609
2609          pendings [ABSPRI (w)][w->pending - 1].w = (W)&pending_w;
```

**Comment by bnoordhuis** [comment]
Thanks for the report, Chris. Can you post the output of `backtrace full` (in gdb, after the crash)?

I suspect that you're running into the file descriptor limit. It's 256 by default on OS X and each child process uses 2 or 3 file descriptors so spawning 100 processes probably hits the limit. Until this is fixed, setting `ulimit -n 16384` should let you work around it.

**Comment by chriso** [comment]
Thanks Ben, upping the descriptor limit fixes the issue.

```
(gdb) backtrace full
#0  0x000000010026a9e9 in clear_pending (loop=0x1004ca3c0, w=0x12530c6d8) at ev.c:2609
No locals.
#1  0x000000010026ba23 in ev_child_stop (loop=0x1004ca3c0, w=0x12530c6d8) at ev.c:3001
No locals.
#2  0x0000000100253cba in uv_close (handle=0x12530c680, close_cb=0x10001b4e4 <node::HandleWrap::OnClose(uv_handle_s*)>) at core.c:141
    udp = (uv_udp_t *) 0x100056a7a
    timer = (uv_timer_t *) 0x1254c6679
    stream = (uv_stream_t *) 0x100000001
    process = (uv_process_t *) 0x12530c680
    async = (uv_async_t *) 0x100000000
#3  0x000000010001b8f9 in node::HandleWrap::Close (args=@0x7fff5fbfed08) at handle_wrap.cc:103
    scope = {
  isolate_ = 0x10081e200, 
  prev_next_ = 0x125801338, 
  prev_limit_ = 0x125802ff0, 
  is_closed_ = false
}
#4  0x00000001000727ae in v8::internal::Builtin_HandleApiCall () at node_crypto.h:166
No symbol table info available.
Previous frame inner to this frame (gdb could not unwind past this frame)
Current language:  auto; currently c
```

**Comment by bnoordhuis** [comment]
Thanks, Chris. Turns out it's a bug in node after all. It's fixed in joyent/node@d5d043f and joyent/node@b07acb3.
